### PR TITLE
✨ Call `access_aws` for all paths and cache

### PR DESCRIPTION
**Pull request by Koncopd**

### CODE REVIEW

**Review by falexwolf — APPROVED** [review]
Looks great! 😄

File: lamindb_setup/core/_settings_storage.py [hunk 1]
```diff
@@ -235,6 +236,12 @@ def root(self) -> UPath:
         if self._root is None:
             # below makes network requests to get credentials
             self._root = create_path(self._root_init, access_token=self.access_token)
+        elif getattr(self._root, "protocol", "") == "s3":
+            # this is needed to be sure that the root always have nonexpired credentials
```
Comment (Koncopd) [inline lamindb_setup/core/_settings_storage.py:240]: to fix: has

Comment (falexwolf) [reply lamindb_setup/core/_settings_storage.py:240]: So, this cannot yet be merged? 🤔

Comment (Koncopd) [reply lamindb_setup/core/_settings_storage.py]: Not it can be.

Comment (falexwolf) [reply lamindb_setup/core/_settings_storage.py]: Great, in that case I'll merge and start playing with it!

**Pull request merged**